modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -327,6 +327,27 @@ func (q Keeper) PacketAcknowledgements(c context.Context, req *types.QueryPacket
 	acks := []*types.PacketState{}
 	store := prefix.NewStore(ctx.KVStore(q.storeKey), []byte(host.PacketAcknowledgementPrefixPath(req.PortId, req.ChannelId)))
 
+	// if a list of packet sequences is provided then query for each specific ack and return a list <= len(req.PacketCommitmentSequences)
+	// otherwise, maintain previous behaviour and perform paginated query
+	for _, seq := range req.PacketCommitmentSequences {
+		acknowledgementBz, found := q.GetPacketAcknowledgement(ctx, req.PortId, req.ChannelId, seq)
+		if !found || len(acknowledgementBz) == 0 {
+			continue
+		}
+
+		ack := types.NewPacketState(req.PortId, req.ChannelId, seq, acknowledgementBz)
+		acks = append(acks, &ack)
+	}
+
+	if len(acks) > 0 {
```

Comment:
One last change

```suggestion
	if len(req.PacketCommitmentSequences) > 0 {
```

I think we should return an empty array if none of the commitments provided had an acknowledgement written as opposed to returning all acknowledgements. That might be surprising behaviour

Reply:
Great point! That scenario skipped my mind.
Updated via Github UI suggestion